be/src/exec/vectorized/aggregate/aggregate_blocking_node.cpp

```diff
@@ -62,7 +67,16 @@ Status AggregateBlockingNode::open(RuntimeState* state) {
             if (_aggregator->is_none_group_by_exprs()) {
                 _aggregator->compute_single_agg_state(chunk->num_rows());
             } else {
-                _aggregator->compute_batch_agg_states(chunk->num_rows());
+                if (group_by_with_limit) {
+                    size_t zero_count = SIMD::count_zero(_aggregator->streaming_selection().data(), chunk->num_rows());
+                    if (zero_count == chunk->num_rows()) {
+                        _aggregator->compute_batch_agg_states(chunk->num_rows());
```

Comment:
Would better make the `chunk->num_rows()` to a variable

Reply:
done